esp/src/src-react/components/ECLPlayground.tsx

```diff
@@ -201,26 +204,38 @@ const ECLEditorToolbar: React.FunctionComponent<ECLEditorToolbarProps> = ({
     const [queryNameErrorMsg, setQueryNameErrorMsg] = React.useState("");
     const [showSubmitBtn, setShowSubmitBtn] = React.useState(true);
 
-    const submitWU = React.useCallback(async () => {
-        const wu = await Workunit.create({ baseUrl: "" });
-
-        await wu.update({ QueryText: editor.ecl() });
-        await wu.submit(cluster);
-
-        wu.watchUntilComplete(changes => {
-            setWuState(wu.State);
+    const playgroundResults = React.useCallback((wu, action = "submit") => {
+        setWuState(wu.State);
+        if (document.location.hash.includes("play")) {
             if (wu.isFailed()) {
                 pushUrl(`/play/${wu.Wuid}`);
                 setWorkunit(wu);
                 displayErrors(wu, editor);
                 setOutputMode(OutputMode.ERRORS);
             } else if (wu.isComplete()) {
                 pushUrl(`/play/${wu.Wuid}`);
+                if (action === "publish") {
+                    wu.publish(queryName);
+                    setWuState("Published");
+                }
                 setWorkunit(wu);
                 setOutputMode(OutputMode.RESULTS);
             }
-        });
-    }, [cluster, editor, setOutputMode, setWorkunit]);
+        } else {
+            if (wu.isFailed() || wu.isComplete()) {
```

Comment:
FYI `isComplete` is true, even when the WU failed.  So no need for the `isFailed` call.